### PR TITLE
Stream execution progress to the action page

### PR DESCRIPTION
**Pull request by bduffany**
Starting with the action page mostly as a proof of concept because it's easier to test than workflows and is maybe somewhat useful for larger actions. It will be more interesting to show this info on the queued invocation page (for workflow / remote bazel invocations).

Screen recording of an execution with an artificial 8s delay at the beginning of ExecuteTaskAndStreamResults + 750ms delay when transitioning between each execution state (otherwise the updates go by too quickly...) (also the "Unknown" execution status should probably be "Queued", need to fix that separately)

https://github.com/buildbuddy-io/buildbuddy/assets/2414826/d76dd269-d966-48a9-b51e-e30696f142be

**Related issues**: N/A

### CODE REVIEW

File: app/invocation/execution_status.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    return "Completed";
+  }
+
+  switch (op.metadata?.stage) {
+    case ExecutionStage.CACHE_CHECK:
+      return "Checking cache";
+    case ExecutionStage.QUEUED:
+      return "Queued";
+    case ExecutionStage.EXECUTING:
+      // Return fine-grained progress using BB-specific enum
```
Comment (jdhollen) [inline app/invocation/execution_status.tsx:48]: nit: for readability, might be better to just break out the `executionState`-based text into its own function and call it directly here--the fall-through is a bit awkward.

File: app/invocation/invocation.tsx [hunk 2]
```diff
@@ -197,6 +197,12 @@ export default class InvocationComponent extends React.Component<Props, State> {
       .catch((error: any) => {
         console.error(error);
         this.setState({ error: BuildBuddyError.parse(error) });
+
+        // To make manual testing easier, support a special 'retry' param that
+        // auto-reloads the page if there is an error.
```
Comment (jdhollen) [inline app/invocation/invocation.tsx:202]: why this one rpc + why this particular behavior?

File: app/invocation/invocation_execution_card.tsx [hunk 3]
```diff
@@ -74,6 +76,12 @@ export default class ExecutionCardComponent extends React.Component<Props, State
         this.fetchUpdatedProgress();
       }
 
+      // To help with manual testing, support a special "openFirst" URL param
+      // which automatically navigates to the first execution that we fetched.
```
Comment (jdhollen) [inline app/invocation/invocation_execution_card.tsx:80]: Probably need to add a list of these somewhere (whether by an enum in code or just a list in a markdown file somewhere)--the way that they're placed right now means me and you will be the only people that know about them.

**Comment by jdhollen** [comment]
One extra thing--might be worth putting a tiny little spinner next to the status or something (until it's completed, obviously)?  As it is it's not obvious that the text will potentially update.

I don't _particularly_ like using a spinner, just think there should be something.

**Review by jdhollen — APPROVED** [review]

**Comment by bduffany** [comment]
> One extra thing--might be worth putting a tiny little spinner next to the status or something (until it's completed, obviously)? As it is it's not obvious that the text will potentially update.

> I don't particularly like using a spinner, just think there should be something.

I agree, I was thinking of maybe a pulsing blue dot but something tells me it's going to look a little weird. Will play around with this in a separate PR

**Pull request merged**